contracts/contracts/coordination/FlateRateFeeModel.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "./IFeeModel.sol";
+import "../../threshold/IAccessControlApplication.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/**
+* @title FlatRateFeeModel
+* @notice FlateRateFeeModel
+*/
+contract FlatRateFeeModel is IFeeModel {
+
+    IERC20 public immutable currency;
+    uint256 public immutable feeRatePerSecond;
+    IAccessControlApplication public immutable stakes;
+
+    constructor(IERC20 _currency, uint256 _feeRatePerSecond, address _stakes){
+        currency = _currency;
+        feeRatePerSecond = _feeRatePerSecond;
+        stakes = IAccessControlApplication(_stakes);
+    }
+
+    function getRitualInitiationCost(
+        address[] calldata providers,
+        uint32 duration
+    ) external view returns(uint256){
+        uint256 size = providers.length;
+        require(duration > 0, "Invalid ritual duration");
+        require(size > 0, "Invalid ritual size");
+        return feeRatePerSecond * size * duration;
```

Comment:
the proposal contains a default `duration` fixed at 182.5 days for initialization 
and then top ups are fixed at 30 days (payable every 30 days) to keep the availability horizon 182.5 days into the future

Reply:
Is this something you would expect to be hardcoded at the smart contract level?

Reply:
I'm opening an issue to track top-ups https://github.com/nucypher/nucypher-contracts/issues/93

Reply:
No not hardcoded, as ofc we may need to change these. Just wanted to note the likely genesis defaults, pending feedback. However there is a small question of trust implications post-v7.0.0 – ideally the Threshold DAO would control sponsorship parameters like these